### PR TITLE
Maddy/newline overrides edit data

### PR DESCRIPTION
**Pull request by MaddyDev**
overrideCellFn is called and the value returned is set as defaultValue of the input text box and it returns value with /r/n. This default value is serialized which strips off the /r/n and shows the string value on UI. And when users just tabs out, the hasChanged event checks the defaultValue and the value inside the input text box and causes the update to trigger. 

Updated the overrideCellFn to replace /r/n with space and this stops the update call when the text isn't changed.

### CODE REVIEW

**Comment by Charles-Gagnon** [comment]
This stops one issue - that is moving out of the edit box (even without changing anything) will commit the change and remove the newlines. But there's still the original issue reported in #1303 where if you edit the value then the new value will be saved without the newlines - so this is still broken.

**Comment by Charles-Gagnon** [comment]
So it looks like the main issue is in node_modules\slickgrid\slick.editors.js. We set the value of the input text box to the value passed in - but setting the value property on the JQuery element is stripping it of the newlines. 

I tried playing around with fixing that but wasn't able to get anything that worked fully. Replacing the newlines with encodings such as &#10; (\n) kind of worked but there's some other display issues with that.

@anthonydresser - Any ideas from your end?

In the end the behavior with Maddy's fix is at least slightly better than current behavior since we won't be deleting newlines without any changes. But this is still a pretty big issue even with that fix so we should definitely try to fix it so the behavior is completely correct.

**Comment by MaddyDev** [comment]
@Charles-Gagnon yes, I tried different ways to address it and found no clean fix to replicate what SSMS is doing. The changes here address only when the text isn't changed. I emailed about this to Karl and for May, we agreed to only address this part.

File: src/sql/workbench/parts/grid/views/editData/editData.component.ts [hunk 1]
```diff
@@ -177,9 +177,9 @@ export class EditDataComponent extends GridParentComponent implements OnInit, On
 		this.overrideCellFn = (rowNumber, columnId, value?, data?): string => {
 			let returnVal = '';
 			if (Services.DBCellValue.isDBCellValue(value)) {
-				returnVal = value.displayValue;
+				returnVal = value.displayValue.replace(/(\r\n|\n|\r)/g, ' ');
```
Comment (alanrenmsft) [inline src/sql/workbench/parts/grid/views/editData/editData.component.ts]: we should introduce a method to handle the line break to space conversion logic, and please add comments explaining why this is added

Comment (Charles-Gagnon) [reply src/sql/workbench/parts/grid/views/editData/editData.component.ts]: I'd update #1303 with this information too and then we should definitely have that as a fix we take for June.

**Pull request merged**